### PR TITLE
fix: update `@bundled-es-modules/cookie` to 2.0.1

### LINKED ISSUE #2308: CVE-2024-47764 vulnerability from the `cookie` package
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Browsers

_No response_

### Reproduction repository

https://github.com/boxwise/boxtribute

### Reproduction steps

The latest msw release (<= 2.4.9) has a transient dependency on several versions of the cookie library that are < 7.0.0, which  has a vulnerability (https://github.com/advisories/GHSA-pxg6-pf52-xh8x).

### Current behavior

You are referencing it via 
* `@bundled-es-modules/cookie@2.0.0` which has not been updated (wrapper appears stale - may be possible to just reference `cookie` directly now?)
* `engine.io@6.5.4`, which has an open PR to fix: https://github.com/socketio/socket.io/pull/5205

While it's unlikely to impact msw given your use-case, it will force the dependency to a lower version for others that use it in production-facing code.

### Expected behavior

No security warnings

### PR DESCRIPTION
**Pull request by kettanaito**
- Fixes #2308

## Blockers

- [x] https://github.com/bundled-es-modules/cookie/pull/3
- [ ] https://github.com/bundled-es-modules/cookie/pull/4
- [x] Bump the version of `@bundled-es-modules/cookie` once https://github.com/bundled-es-modules/cookie/pull/3 is released.

### CODE REVIEW

File: src/core/utils/request/getRequestCookies.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
-import cookieUtils from '@bundled-es-modules/cookie'
+import * as cookieUtils from 'cookie'
 import { cookieStore } from '../cookieStore'
 
+function parseCookies(input: string): Record<string, string> {
+  const parsedCookies = cookieUtils.parse(input)
+  const cookies: Record<string, string> = {}
+
+  for (const cookieName in parsedCookies) {
+    if (typeof parsedCookies[cookieName] !== 'undefined') {
```
Comment (kettanaito) [inline src/core/utils/request/getRequestCookies.ts:9]: Account for `cookie@1.0` annotating the parsing result as `Record<string, string | undefined>`. Skip the cookies with undefined values.

**Review by marklai1998 — APPROVED** [review]

**Comment by kettanaito** [comment]
I've opened a PR to the bundled cookie package to have release automation to release fixes faster. https://github.com/bundled-es-modules/cookie/pull/4

**Comment by curtdept** [comment]
Im confused, why go through bundle es proxy when you can just directly use the cookie lib? The lag time on this update through the proxy is crazy.

**Comment by kettanaito** [comment]
The fix has been released in `@bundled-es-modules/cookie` in 2.0.1. Bumped the version range in MSW. Will merge once the CI is passing.

**Comment by kettanaito** [comment]
## Released: v2.6.2 🎉

This has been released in v2.6.2!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.6.2)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.6.2)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**